mlir-tensorrt/test/Conversion/TensorRTRuntimeToExecutor/tensorrt-runtime-to-executor.mlir

```diff
@@ -133,4 +133,93 @@ func.func @main(%arg0: memref<1x3x256x256xf32, #executor.memory_type<device>>) -
 // CHECK: %[[v36:.*]] = builtin.unrealized_conversion_cast %[[v35]] : !executor.table<!executor.ptr<device>, !executor.ptr<device>, i64, i64, i64, i64, i64, i64, i64, i64, i64> to memref<?x?x?x?xf32, #executor.memory_type<device>>
 // CHECK: cuda.stream.sync %[[v8]] : !cuda.stream
 // CHECK: return %[[v36]] : memref<?x?x?x?xf32, #executor.memory_type<device>>
-// CHECK: }
\ No newline at end of file
+// CHECK: }
+
+// -----
+
+tensorrt.module @trt_funcs {
+  func.func @my_func() attributes {tensorrt.engine = dense<0> : vector<10xi8>} {
+    return
+  }
+}
+
+func.func @main(%arg0: memref<?x?x?xf32, #executor.memory_type<device>> {tensorrt.shape_profile = #executor.dim_bounds<min = [1, 1, 2], max = [1, 4, 2]>}, %arg1: memref<?x?xi32, #executor.memory_type<device>> {tensorrt.shape_profile = #executor.dim_bounds<min = [1, 1], max = [1, 4]>}) -> (memref<?x?x256xf32, #executor.memory_type<device>> {tensorrt.shape_profile = #executor.dim_bounds<min = [1, 2, 256], max = [1, 5, 256]>}, memref<?x256x64x64xf32, #executor.memory_type<device>> {tensorrt.shape_profile = #executor.dim_bounds<min = [1, 256, 64, 64], max = [1, 256, 64, 64]>}) attributes {executor.function_metadata = #executor.func_meta<[memref<?x?x?xf32, #executor.memory_type<device>> {#executor.dim_bounds<min = [1, 1, 2], max = [1, 4, 2]>}, memref<?x?xi32, #executor.memory_type<device>> {#executor.dim_bounds<min = [1, 1], max = [1, 4]>}], [memref<?x?x256xf32, #executor.memory_type<device>> {#executor.dim_bounds<min = [1, 2, 256], max = [1, 5, 256]>}, memref<?x256x64x64xf32, #executor.memory_type<device>> {#executor.dim_bounds<min = [1, 256, 64, 64], max = [1, 256, 64, 64]>}], num_output_args = 0>} {
```

Comment:
Tests should strive to contain` the *minimum* amount of IR/code to exercise the pass we are testing here. 

So all the `tensorrt.shape_profile`, etc should be removed. Think about it this way: If we update or change the IR textual assembly format for `#executor.dim_bounds`, we would need to update this test, but that component is not even needed here.

Similarly, the `bufferization.clone` should be removed.

Since this particular test does not need to test conversion of `trtrt.compile` or `cuda.get_global_stream`, you could just add additional arguments `%argX: !trtrt.context` and `%argY: !cuda.stream` to further simplify the test and the amount of `CHECK` statements that need to be written below.

Reply:
Excellent point. I have simplified the IR now.